plugins/mission_raw/include/plugins/mission_raw/mission_raw.h

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <functional>
+#include "plugin_base.h"
+
+namespace dronecode_sdk {
+
+class MissionRawImpl;
+class System;
+
+/**
+ * @brief The MissionRaw class enables direct direct access to MAVLink
+ *        mission items.
+ */
+class MissionRaw : public PluginBase {
+public:
+    /**
+     * @brief Constructor. Creates the plugin for a specific System.
+     *
+     * The plugin is typically created as shown below:
+     *
+     *     ```cpp
+     *     auto mission_raw = std::make_shared<MissionRaw>(system);
+     *     ```
+     *
+     * @param system The specific system associated with this plugin.
+     */
+    explicit MissionRaw(System &system);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~MissionRaw();
+
+    /**
+     * @brief Possible results returned for mission requests.
+     */
+    enum class Result {
+        UNKNOWN, /**< @brief Unknown error. */
+        SUCCESS, /**< @brief %Request succeeded. */
+        ERROR, /**< @brief Error. */
+        BUSY, /**< @brief %Vehicle busy. */
+        TIMEOUT, /**< @brief Request timed out. */
+        INVALID_ARGUMENT, /**< @brief Invalid argument. */
+        NO_MISSION_AVAILABLE, /**< @brief No mission available on system. */
+        CANCELLED /**< @brief Mission upload or download has been cancelled. */
+    };
+
+    /**
+     * @brief Gets a human-readable English string for an MissionRaw::Result.
+     *
+     * @param result Enum for which string is required.
+     * @return Human readable string for the MissionRaw::Result.
+     */
+    static const char *result_str(Result result);
+
+    /**
+     * @brief Mission item identical to MAVLink MISSION_ITEM_INT.
+     */
+    struct MavlinkMissionItemInt {
```

Comment:
No support for MISSION_ITEM?

Reply:
Nope. The SDK only does the new shiny things.